Git-guide,-TODO-'n'-notTODO.md

```diff
@@ -1,42 +1,45 @@
+  6. Make a pull request
+- **Do not merge branches, or pull without --rebase**
+  - Merges create extra merge commits we don't need and want in our main repo
+  - Rebase your changes instead. So when you want to make your pull request up-to-date with master, use `git pull --rebase origin master` rather than a simple `git pull origin master`
+  - If you rebased your changes, you will want to use `git push --force-with-lease` to push your changes to your fork
+- When you want to test someone's changes
+  - `git branch <branch name>` and `git branch <branch name> -u <remote>/<branch name>`
+  - Make sure that you are on the root of that branch. Otherwise DO NOT push your merge commit!
 
 ## How to rebase your commits
+
 Unless you have more than 20 commits in one pull request, don't squash normal commits down until asked. This makes it easier to check your code for problems if any occur. When it is agreed that your pull request is worth adding to the project, then you may be asked to rebase.
 
 For a tutorial on rebasing look [here](https://help.github.com/articles/about-git-rebase/).
 
 In order to get the rebase changes to take effect you will have to force push to your fork of supertux. **Only force push to your own fork of Supertux**.
 
 ## Other things you should respect
-* Never force-push to a shared branch.
-* Don't constantly ask other people to update their content.
-* Use only constructive criticism, as being rude or acting annoyed could scare off a potential developer!
-* Don't beg anyone to teach you a something. Instead, ask them for guidance if you encounter a problem when following an online tutorial
-* Don't revert other people's work unless you're told to do so.
-* Don't push erroneous commits to master.
-* Don't make with vague suggestions. Always be specific of what you need.
-* We don't need any visionaries. Being productive in suggestions those someone could do is dangerous here.
-* When you make a big change, push it to your forked GitHub repo. and make a pull request.
-* Owners: Try not to push more than 3 commits straight to master at once. For small changes it should be rebased first and for big changes it should be put in a pull request.
\ No newline at end of file
+
+- Never force-push to a shared branch
+- Don't constantly ask other people to update their content
+- Use only constructive criticism, as being rude or acting annoyed could scare off a potential developer!
+- Don't beg anyone to teach you something. Instead, ask them for guidance if you encounter a problem when following an online tutorial
+- Don't revert other people's work unless you're told to do so
+- Don't push erroneous commits to master
+- Don't be too vague, try to be as specific as possible
+- We don't need any visionaries. Being productive in suggestions those someone could do is dangerous here
```

Comment:
"Being productive in suggestions those someone could do is dangerous here" <-- I still don't understand that sentence.

Reply:
I'll remove that entirely.